pkg/kubectl/cmd/util/factory_client_access.go

```diff
@@ -392,8 +394,10 @@ func (f *ring0Factory) SuggestedPodTemplateResources() []schema.GroupResource {
 	}
 }
 
-func (f *ring0Factory) Printer(mapping *meta.RESTMapping, options kubectl.PrintOptions) (kubectl.ResourcePrinter, error) {
-	return kubectl.NewHumanReadablePrinter(options), nil
+func (f *ring0Factory) Printer(mapping *meta.RESTMapping, options printers.PrintOptions) (printers.ResourcePrinter, error) {
+	p := printers.NewHumanReadablePrinter(options)
+	printersinternal.AddHandlers(p)
```

Comment:
I'm reading top to bottom, so maybe this becomes clear, but I'm really surprised to find some of registration happening inside a factory method that could be called multiple times.